### PR TITLE
fix issue 16360 - DMD fails to inline functions that contain a type

### PR DESCRIPTION
**Pull request by ghost**
It will fail less with this change.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Review by WalterBright — DISMISSED** [review]
see my comments

**Comment by Basile-z** [comment]
I've addressed the review comments but please can you put the WIP label ?

- coverage results are wrong. I think this is because the aggregates are not copied. 
- i'd like to see why even a simple function inside a nested struct blocks inlining. Currently the enhancement is very minor and will have no impact on range code.

**Comment by Basile-z** [comment]
No this is good actually. The coverage thing is inherent to inlining.

**Comment by Basile-z** [comment]
yes even for

```d
pragma(inline, true)
int foo(float s)
{
    auto bt = 3;
    return bt;
}

void main()
{
    auto a = foo(1.0f);
} 
```

you got `00000` for the content of `foo`. There was no problem sorry.

File: test/fail_compilation/b16360.d [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+REQUIRED_ARGS: -inline
+
+TEST_OUTPUT:
+---
+fail_compilation/b16360.d(12): Error: function `b16360.foo` cannot inline function
+fail_compilation/b16360.d(25): Error: function `b16360.bar` cannot inline function
+---
+*/
+
+pragma(inline, true)
```
Comment (Basile-z) [inline test/fail_compilation/b16360.d]: @ibuclaw is pragma inline supported by GDC too ?

Comment (ibuclaw) [reply test/fail_compilation/b16360.d]: Yes, but this test won't fail when compiled by gdc.

Comment (Basile-z) [reply test/fail_compilation/b16360.d]: So I must add a version() for LDC and GDC is suppose ?

Comment (ibuclaw) [reply test/fail_compilation/b16360.d]: I'm sure they more than capable of omitting this test file when mirroring it into their repository.

**Comment by Basile-z** [comment]
please @thewilsonator can you remove the WIP label ?

**Comment by thewilsonator** [comment]
Done.

**Comment by thewilsonator** [comment]
This good to go?

**Comment by Basile-z** [comment]
> This good to go?

If you asked me yes, but  @WalterBright  hasn't approved, which seems required to cancel his previous review.

**Comment by thewilsonator** [comment]
I'll merge tomorrow morning if he hasn't said anything.

**Pull request merged**